src/software/backend/simulation/simulator_robot_test.cpp

```diff
@@ -3,39 +3,840 @@
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(-1, 0)).length(), 0.001);
+
+    simulator_robot->enableAutochip(5);
+    simulator_robot->disableAutochip();
+
+    // Simulate for 1/2 second
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    EXPECT_LT((getDribblingPoint(simulator_robot) - simulator_ball->position()).length(),
+              0.1);
+}
+
+TEST_F(SimulatorRobotTest, test_enabling_autokick_disables_autochip)
+{
+    Robot robot(0, Point(0, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball(Point(0.25, 0), Vector(-1, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(-1, 0)).length(), 0.001);
+
+    simulator_robot->enableAutochip(1);
+    simulator_robot->enableAutokick(5);
+
+    // Simulate for 1/2 second
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(4, 0)).length(), 0.001);
+}
+
+TEST_F(SimulatorRobotTest, test_enabling_autochip_disables_autokick)
+{
+    Robot robot(0, Point(0, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball(Point(0.25, 0), Vector(-1, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(-1, 0)).length(), 0.001);
+
+    simulator_robot->enableAutokick(5);
+    simulator_robot->enableAutochip(2);
+
+    // Simulate for 1/2 second
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    EXPECT_GT(simulator_ball->velocity().x(), 1);
+    EXPECT_LT(simulator_ball->velocity().x(), 2);
+}
+
+TEST_F(SimulatorRobotTest, test_ball_does_not_bounce_off_front_of_robot_when_dribbler_on)
+{
+    Robot robot(0, Point(0, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball(Point(0.25, 0), Vector(-3, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(-3, 0)).length(), 0.001);
+
+    // We use an arbitrarily large number here for speed
+    simulator_robot->setDribblerSpeed(10000);
+
+    // Simulate for 2 seconds
+    for (unsigned int i = 0; i < 120; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    // Make sure we haven't caused the robot to move significantly by dribbling
+    EXPECT_LT((simulator_robot->position() - Point(0, 0)).length(), 0.003);
```

Comment:
redundant Point(0,0)

Reply:
Intentionally redundant for explicitness, so it matches the initial state of the robot specified at the start of the test